plugins/modules/sda_host_port_onboarding_workflow_manager.py

```diff
@@ -0,0 +1,3010 @@
+            self.msg = (
+                "An error occurred while retrieving Port Assignment Details: '{0}' using SDA - "
+                "'get_port_assignments' API call: {1}".format(get_port_assignments_params, str(e))
+            )
+            self.fail_and_exit(self.msg)
+
+    def compare_port_assignments(self, get_port_assignments_params, requested_port_assignment_details):
+        """
+        Compares existing port assignments with requested port assignments to determine required actions.
+        Args:
+            get_port_assignments_params (dict): Parameters for querying existing port assignments.
+            requested_port_assignment_details (list): List of requested port assignment details.
+        Returns:
+            tuple: A tuple containing three lists:
+                - create_port_assignments (list): List of port assignments to be created.
+                - update_port_assignments (list): List of port assignments to be updated.
+                - no_update_port_assignments (list): List of port assignments that do not need updates.
+        Description:
+            This method compares existing port assignments retrieved from Cisco Catalyst Center with the requested
+            port assignment details. It categorizes the port assignments into those that need to be created, updated,
+            or do not require any updates. The method logs relevant information and returns the categorized lists.
+        """
+        existing_port_assignment_details = self.get_port_assignments(get_port_assignments_params)
+
+        self.log("Existing Port assignments: {0}".format(existing_port_assignment_details), "DEBUG")
+        self.log("Requested Port assignments: {0}".format(requested_port_assignment_details), "DEBUG")
+
+        create_port_assignments = []
+        update_port_assignments = []
+        no_update_port_assignments = []
+
+        # Convert the requested_port_assignment_details to a dictionary for quick lookup
+        requested_ports_dict = {port['interface_name']: port for port in requested_port_assignment_details}
+
+        if not existing_port_assignment_details:
+            self.log("Port assignments that need to be CREATED: {0} - {1}".format(len(create_port_assignments), create_port_assignments), "DEBUG")
+            create_port_assignments.extend(requested_ports_dict.values())
+            return create_port_assignments, update_port_assignments, no_update_port_assignments
+
+        # Iterate over existing ports to find matches and differences
+        for existing_port in existing_port_assignment_details:
+            interface_name = existing_port['interfaceName']
+            # If the interface exists in both, compare fields
+            if interface_name in requested_ports_dict:
+                requested_port = requested_ports_dict[interface_name]
+
+                # Compare the relevant fields (excluding id, fabricId, networkDeviceId)
+                has_diff = False
```

Comment:
Can we move this logic into an API for better readability?

`            has_diff = self.check_for_differences(existing_port, requested_port)
`
```
def check_for_differences(self, existing_port, requested_port):
    """
    Compares the existing port with the requested port for differences.

    Args:
        existing_port (dict): The existing port details.
        requested_port (dict): The requested port details.

    Returns:
        bool: True if there are differences, False otherwise.
    """
    comparison_fields = [
        ("interfaceName", "interface_name"),
        ("connectedDeviceType", "connected_device_type"),
        ("authenticateTemplateName", "authentication_template_name"),
        ("dataVlanName", "data_vlan_name"),
        ("voiceVlanName", "voice_vlan_name"),
        ("interfaceDescription", "interface_description"),
        ("securityGroupName", "security_group_name")
    ]

    for existing_field, requested_field in comparison_fields:
        if existing_field == "authenticateTemplateName":
            if existing_port.get("authenticateTemplateName") == "No Authentication" and not requested_port.get("authentication_template_name"):
                continue
            if not existing_port.get("authenticateTemplateName") and requested_port.get("authentication_template_name") == "No Authentication":
                continue

        if existing_field == "interfaceDescription":
            if existing_port.get("interfaceDescription") == "" and not requested_port.get("interface_description"):
                continue
            if not existing_port.get("interfaceDescription") and requested_port.get("interface_description") == "":
                continue
            if existing_port.get("interfaceDescription") and not requested_port.get("interface_description"):
                continue

        # Check if fields exist in both and compare their values
        existing_value = existing_port.get(existing_field)
        requested_value = requested_port.get(requested_field)

        if existing_value != requested_value:
            return True

    return False
```